### ISSUE TITLE
1.0.3: App shows Animal entry and Animal left but timing is not as expected.

### ISSUE THREAD

**Issue by akshayvernekar** [akshayvernekar/Brilliant_pad_issue_management#13]
Can you clarify what is happening?

**Comment by akshayvernekar** [comment]
This is the limitation with MQTT/AWS IOT broker. 
We are currently using MQTT with QOS of 0 which means that there is no guarantee that the message sent will be delivered . Thats why sometimes we are seeing Animal left and dont see Animal entered event.
We can switch to QOS 1 , in QOS 1 message is guaranteed to be delivered but it might sent duplicate messages .So you can expect to see events sent twice , but the app or server can filter the messages received based on the PACKET ID.
QOS 2 will be perfect but messages will be delivered slow and also AWS IOT doesnt support QOS 2.
PRAVEENA: Please make the changes on the server side to change the QOS to 1 .

**Comment by alanjcook** [comment]
Sometimes the animal entry does not show up on app -- this was shown several times in the video.

**Comment by akshayvernekar** [comment]
Yes I checked there were no changes made with regard to MQTT or Animal detection . with QOS 0 not every message is guaranteed to be delivered but QOS 1 will guarantee delivery .

**Comment by alanjcook** [comment]
Praveena and Akshay reviewed this yesterday. Please review firmware code around lookin to identify potential cause problems.

**Comment by alanjcook** [comment]
sometimes app does not show animal detected.

**Comment by akshayvernekar** [comment]
I will discuss with Praveena regarding this.

**Comment by akshayvernekar** [comment]
@cumulations  need to change the MQTT setting to QOS1

**Comment by alanjcook** [comment]
AKSHAY: “Delayed animal entry/exit”  seems to happen after system appears to be “overloaded” with many commands in small amount to time. This feels related to situations when LOOK IN stops working — the machine indicates receipt of a command from the server (by cyclopsing all three LEDs) but does not send the image back to the cloud. 

Why do some machines stop responding to Look In Commands from time to time? 

Why is Animal Entry/Exit appearing in log at “wrong time” a long time after events occur?

Changing MQTT QOS will help ensure all commands get to/from machine, but will not solve machine not properly responding to commands or not sending info to cloud.

**Comment by akshayvernekar** [comment]
Hi Alan, this issue is regarding events not coming in the correct order, this has to do with QOS as some events might have been lost so you might be seeing two animal entry events with a lost "exit" event . Also MQTT doesnt guarantee delivery  of commands in the same order they are sent .

**Comment by akshayvernekar** [comment]
@alanjcook added your comment to https://github.com/akshayvernekar/Brilliant_pad_issue_management/issues/10
